src/Stripe.net/Infrastructure/JsonConverters/STJAnyOfConverter.cs

```diff
@@ -0,0 +1,96 @@
+#if NET6_0_OR_GREATER
+namespace Stripe.Infrastructure
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text.Json;
+    using System.Text.Json.Serialization;
+
+    /// <summary>
+    /// Converts a <see cref="IAnyOf"/> to and from JSON.
+    /// </summary>
+    public class STJAnyOfConverter : JsonConverter<object>
+    {
+        /// <summary>
+        /// Reads the JSON representation of the object.
+        /// </summary>
+        /// <param name="reader">The <see cref="Utf8JsonReader"/> to read from.</param>
+        /// <param name="typeToConvert">Type of the object.</param>
+        /// <param name="options">The calling serializer's options.</param>
+        /// <returns>The object value.</returns>
+        public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+
+            object o = null;
+
+            // Try to deserialize with each possible type
+            var elem = JsonElement.ParseValue(ref reader);
+            foreach (var type in typeToConvert.GenericTypeArguments)
+            {
+                try
+                {
+                    o = JsonSerializer.Deserialize(elem.GetRawText(), type, options);
+
+                    // If deserialization succeeds, stop
+                    break;
+                }
+                catch (JsonException)
+                {
+                    // Do nothing, just try the next type
```

Comment:
Is this how newtonsoft works today as well?

Reply:
yep!  https://github.com/stripe/stripe-dotnet/blob/4d0c154eb7d4a399d21504f1b9ce1f5be14e8cc6/src/Stripe.net/Infrastructure/JsonConverters/AnyOfConverter.cs#L80

for a lot of these converters, I copied the NS one and changed the objects/methods to their equivalents.